git/index.go

```diff
@@ -199,7 +200,26 @@ func (g *Index) AddStage(c *Client, path IndexPath, mode EntryMode, s Sha1, stag
 			entry.Mtime = mtime
 			entry.Fsize = size
 
-			// We found and updated the entry, no need to continue
+			// If replace is true then we search for any entries that
+			//  should be replaced with this one. It could be that there
+			//  are entries with the same SHA-1 hash or there are children
```

Comment:
from the man page, I don't really understand where the bit about the SHA1 comes from

Reply:
From what I observed from comparing the differences between how official git and dgit handle the D/F conflict test case I saw that the -add --replace was expected to remove one of the entries based on nothing that I could see except for the sha-1 hash of an existing item in the staging area. They shared nothing in common except for the contents. Perhaps there is another way that official git is matching them?

Reply:
I don't really know much about how --replace is supposed to work, I'm just going by my understanding of reading the man page.

If it passes the official tests, it's good enough for me.

Reply:
I'm going to spend a bit more time on this and try some more scenarios and re-read the documentation.